MediaWikiKit/MediaWikiKit/MWKHistoryList.m

```diff
@@ -70,6 +70,16 @@ - (void)setPageScrollPosition:(CGFloat)scrollposition onPageInHistoryWithTitle:(
     }];
 }
 
+- (void)setSignificantlyViewedOnPageInHistoryWithTitle:(MWKTitle*)title {
+    if ([title.text length] == 0) {
+        return;
+    }
+    [self updateEntryWithListIndex:title update:^BOOL (MWKHistoryEntry* __nullable entry) {
+        entry.titleWasSignificantlyViewed = YES;
```

Comment:
if you return `NO` here, it won't re-save the list, e.g. if the entry was already marked as significantly viewed.